src/rcc/rec.rs

```diff
@@ -0,0 +1,551 @@
+// #[cfg(all())]. The description is not used. Each bus grouping can be repeated
+// multiple times if needed.
+//
+// As well as busses, peripherals can optionally be preceeded by a conditional
+// compilation attribute. However, this only works for peripherals without
+// kernel clock multiplexers.
+//
+// Peripherals with an individual kernel clock must be marked "kernel clk". If a
+// kernel clock multiplexer is shared between multiple peripherals, all those
+// peripherals must instead be marked with a common "group clk".
+//
+// NOTE: The stm32h5 crate does not have definitions for the clock source variants for registers for
+// any processors other than the STM32H503 at this time. Therefore, peripherals which have
+// selectable clock sources are currently only specified for the H503 (rm0492).
+peripheral_reset_and_enable_control! {
+
+    #[cfg(all())]
+    AHB1, "AMBA High-performance Bus (AHB1) peripherals" => [
+        RamCfg,
+        Crc,
+
+        Gpdma1,
+        Gpdma2
+    ];
+
+    #[cfg(all())]
+    AHB2, "AMBA High-performance Bus (AHB2) peripherals" => [
+        Hash,
+        Gpioh,
+        Gpiod,
+        Gpioc,
+        Gpiob,
+        Gpioa
+    ];
+    #[cfg(feature = "rm0492")]
+    AHB2, "" => [
+        Rng [kernel clk: Rng ccipr5 "RNG"],
+        Adc [group clk: AdcDac(Variant) ccipr5 "ADC/DAC"],
+        Dac12 [group clk: AdcDac]
+    ];
+
+
+    #[cfg(all())]
+    APB1L, "Advanced Peripheral Bus 1L (APB1L) peripherals" => [
+        Crs,
+
+        Tim2, Tim3, Tim6, Tim7
+    ];
+    #[cfg(feature = "rm0492")]
+    APB1L, "" => [
+        I3c1 [kernel clk: I3c1(Variant) ccipr4 "I3C1"],
+
+        I2c1 [kernel clk: I2c1 ccipr4 "I2C1"],
+        I2c2 [kernel clk: I2c2 ccipr4 "I2C2"],
+
+        Usart2 [kernel clk: Usart2(Variant) ccipr1 "USART2"],
+        Usart3 [kernel clk: Usart3(Variant) ccipr1 "USART3"],
+
+
+        Spi2 [kernel clk: Spi2(Variant) ccipr3 "SPI2"],
+        Spi3 [kernel clk: Spi3(Variant) ccipr3 "SPI3"],
+        Opamp,
+        Comp
+    ];
+
+    #[cfg(all())]
+    APB1H, "Advanced Peripheral Bus 1H (APB1H) peripherals" => [
+        Dts
+    ];
+    #[cfg(feature = "rm0492")]
+    APB1H, "" => [
+        Lptim2 [kernel clk: Lptim2(Variant) ccipr2 "LPTIM2"],
+        Fdcan [kernel clk: Fdcan(Variant) ccipr5 "FDCAN"]
+    ];
+
+    #[cfg(all())]
+    APB2, "Advanced Peripheral Bus 2 (APB2) peripherals" => [
+        Tim1
+    ];
+    #[cfg(feature = "rm0492")]
+    APB2, "" => [
+        Usb [kernel clk: Usb ccipr4 "USB"],
+        Usart1 [kernel clk: Usart1(Variant) ccipr1 "USART1"],
+        Spi1 [kernel clk: Spi1(Variant) ccipr3 "SPI1"]
+    ];
+
+    #[cfg(all())]
+    APB3, "Advanced Peripheral Bus 3 (APB3) peripherals" => [
+        Sbs
+    ];
+    #[cfg(feature = "rm0492")]
+    APB3, "" => [
+        I3c2 [kernel clk: I3c2(Variant) ccipr4 "I3C2"],
+        LpTim1 [kernel clk: LpTim1(Variant) ccipr2 "LPTIM1"],
+        LpUart1 [kernel clk: LpUart1(Variant) ccipr3 "LPUART1"]
```

Comment:
There's also the `RTCAPBEN` bit here. On H7 this was enabled by default after reset, so the HAL ignores it. If I understand the RM correctly, on H5 it will need to be enabled to access the RTC registers.

Reply:
Looks like I excluded this, intentionally or not, because the RTC isn't resettable via the RCC. I'm looking for a way to account for that in the macro definition.

Reply:
I updated the macro to handle the cases where the peripheral is not resettable, and added missing peripherals, including the RTC

Reply:
Great that those methods are available now!